### ISSUE TITLE
404 When trying to filter by claimed only on page 2 with organisation search

### ISSUE THREAD

**Issue by Mike-Heneghan** [Mike-Heneghan/ALISS#51]
As per: http://localhost:5000/organisations/search/?q=test&page=2&is_claimed=true

<img width="1214" alt="Screenshot 2019-03-14 at 11 53 32" src="https://user-images.githubusercontent.com/36415632/54354979-d582f180-464f-11e9-9cda-e7a1a4821027.png">

When on the second page of results and the user selects to filter by claimed=true a 404 is returned as above. 

Need to investigate the ways the params are being extracted and what's occuring.

**Comment by Mike-Heneghan** [comment]
Just checked that the filter for organisations handles the published filter without any issues when there is a result.

url:
`http://localhost:5000/organisations/search/?q=test&page=2&is_published=true`

<img width="1253" alt="Screenshot 2019-03-14 at 12 13 38" src="https://user-images.githubusercontent.com/36415632/54356085-a326c380-4652-11e9-8db1-4a3e0ae22eda.png">

Although adding moving to page 3

url:
`http://localhost:5000/organisations/search/?q=test&page=3&is_published=true`

<img width="1205" alt="Screenshot 2019-03-14 at 12 15 00" src="https://user-images.githubusercontent.com/36415632/54356167-d6695280-4652-11e9-9b80-51465bf75180.png">

**Comment by Mike-Heneghan** [comment]
On services search if there are no results with a filter the same error occurs

<img width="1257" alt="Screenshot 2019-03-14 at 12 17 47" src="https://user-images.githubusercontent.com/36415632/54356330-44ae1500-4653-11e9-8bdc-ec97c5019948.png">

**Comment by Mike-Heneghan** [comment]
So the service search and organisation search share the same bug that if the user moves to a page beyond 1 and then applies a filter which has not results the the user is presented a 404.

I could test a solution on the organisation search which we could also apply to the services search.

**Comment by Mike-Heneghan** [comment]
With logging found my approach to solving the issue was failing as the object_list was being reduced to zero after the conditional check.

**Comment by Mike-Heneghan** [comment]
Logged out the queryset when on a page with an invalid page number and saw that there are results in the queryset but they must be associated with different `page_obj`

```
3:36:35 PM web.1           |  test
3:36:36 PM web.1           |  <QuerySet [<Organisation: Visualise Scotland>, <Organisation: Ninth Testing Org>, <Organisation: Banana Unpublished>, <Organisation: Scottish Men’s Shed Association>, <Organisation: Second Testing Org>, <Organisation: Fourth Testing Org>, <Organisation: Eighth Testing Org>, <Organisation: Eleventh Testing Org>, <Organisation: Twelfth Testing Org>, <Organisation: Home Link Family Support Test 2>, <Organisation: Seventh Testing Org>, <Organisation: >, <Organisation: Check>, <Organisation: banana w
3:36:36 PM web.1           |  >  eird>, <Organisation: Carers of East Lothian>, <Organisation: Capability Scotland>, <Organisation: Fifth Testing Org>, <Organisation: Sixth Testing Org>, <Organisation: Unpublished org>, <Organisation: Testing Organisation>, '...(remaining elements truncated)...']>
3:36:36 PM web.1           |  Not Found: /organisations/search/
3:36:37 PM web.1           |  Not Found: /favicon.ico
```

As the page is out of range we have an exception https://docs.djangoproject.com/en/1.11/_modules/django/core/paginator/#EmptyPage

So I was trying to find a way to have a custom exception page although instead I think it would be best to alwways return to page 1 when a filter is applied. 

The first page is allowed to have no results and this avoid the issue.

**Comment by Mike-Heneghan** [comment]
When a filter is applied need to ensure the page number is not kept and is always set back to 1.

**Comment by Mike-Heneghan** [comment]
Need to update the `query_transform` method needs to be updated although not working successfullly at the moment.

**Comment by Mike-Heneghan** [comment]
The scope of this bug changed from when it was originally written. Tried updating the query transform template tag although even when I did there were issues with it always redirecting the user back to page 1. Instead added a `page=None` to the search views `filter` options i.e. a doesn't run the risk of a 404 as it only reorders the data whereas a filter does as it can reduce the data on the second page. 

As this is UI related I don't think it's easily replicated with automated tests with our current approach.
